### PR TITLE
Improve keyword scans

### PR DESCRIPTION
**Pull request by bra1ncramp**
## 🗣 Description ##

This PR makes two changes:
* Outputs the hostname of each instance
* Uses `zgrep` search command. NOTE: `--recursive` is removed since it is not a valid flag for `zgrep`

## 💭 Motivation and context ##
* Providing the instance hostname gives more clarity in the log output, making it more useful in rolled-over logs
* Using `grep` to search for key words on instances will only reveal possible IOCs if the incident happened within the last rollover date for the log files.  Switching to `zgrep` enables us to search inside rolled-over (gzipped) log files increases the thoroughness of the scans, thus improving the utility of this script.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

### CODE REVIEW

File: extras/ioc_scan_by_host.sh
Comment (mcdonnnj) [inline extras/ioc_scan_by_host.sh]: Updated to reflect the change in functionality.

```suggestion
  # Use find-zgrep to search for IOC strings in log files (including
  # gzipped log files), ignoring *.journal files.  We pipe the result
  # into another grep process that uses the --invert-match grep flag
  # to exclude matches (e.g., from sudo.log) that contain our grep
  # command (e.g. sudo.log).
```

**Review by mcdonnnj — APPROVED** [review]
This seems reasonable and LGTM. I have one suggestion just to update a comment to reflect the change in functionality.

File: extras/ioc_scan_by_host.sh
Comment (dav3r) [inline extras/ioc_scan_by_host.sh]: Remove extraneous comma (I know it predates this PR):
```suggestion
  # to exclude matches (e.g. from sudo.log) that contain our zgrep
```

File: extras/ioc_scan_by_host.sh
Comment (dav3r) [inline extras/ioc_scan_by_host.sh]: @bra1ncramp When we discussed this in chat, we said that wanted to exclude matches from previous runs of this script (when our command was `grep` and not `zgrep`), but since that is just a short-term situation, I think it makes more sense to just change this for the long term:
```suggestion
    --parameters="command='hostname; for i in ${iocList[*]}; do sudo find /var/log -type f -not -name \*\.journal -exec zgrep --ignore-case \$i {} \; | grep --invert-match -- zgrep\ --ignore-case | echo \$(wc --lines) found for \$i; done'"
```

Otherwise, we will just confuse our future selves and future maintainers of this code.

**Review by dav3r — CHANGES_REQUESTED** [review]
I have a couple of small requests:

File: extras/ioc_scan_by_host.sh
Comment (jsf9k) [inline extras/ioc_scan_by_host.sh]: Have you tested this code?  The `grep --invert-match -- --ignore-case\ --recursive\ ` bit is supposed to filter out any lines where the IOCs appear just because you ran `ioc_scan_by_host.sh`.  Hence I _think_ it should be:
```suggestion
    --parameters="command='hostname; for i in ${iocList[*]}; do sudo find /var/log -type f -not -name \*\.journal -exec zgrep --ignore-case \$i {} \; | grep --invert-match -- zgrep\ --ignore-case | echo \$(wc --lines) found for \$i; done'"
```

Otherwise it won't match the `zgrep --ignore-case \$i` jazz.  Or am I missing something?

**Review by jsf9k — CHANGES_REQUESTED** [review]

File: extras/ioc_scan_by_host.sh
Comment (dav3r) [reply extras/ioc_scan_by_host.sh]: See [my comment](https://github.com/cisagov/ioc-scanner/pull/53#discussion_r1379097993) 6 minutes before yours. 😆

Comment (jsf9k) [reply extras/ioc_scan_by_host.sh]: See https://github.com/cisagov/ioc-scanner/pull/53#discussion_r1379097993.

Comment (bra1ncramp) [reply extras/ioc_scan_by_host.sh]: That's fair. I had the same exact thought, to be honest.

**Review by dav3r — APPROVED** [review]
👍  Good stuff!

**Review by jsf9k — APPROVED** [review]

**Review by mcdonnnj — CHANGES_REQUESTED** [review]
@cisagov/team-ois I'm a bit hesitant about the version bump given that this PR does not change the Python package functionality at all. Thoughts?

**Comment by dav3r** [comment]
> @cisagov/team-ois I'm a bit hesitant about the version bump given that this PR does not change the Python package functionality at all. Thoughts?

I was on the fence about that also, but I told @bra1ncramp to do it.  Since we are changing the way an included script works with this PR, I felt it was warranted, but honestly, I'm fine either way if you and @jsf9k want to overrule that.

**Comment by jsf9k** [comment]
> > @cisagov/team-ois I'm a bit hesitant about the version bump given that this PR does not change the Python package functionality at all. Thoughts?
> 
> I was on the fence about that also, but I told @bra1ncramp to do it. Since we are changing the way an included script works with this PR, I felt it was warranted, but honestly, I'm fine either way if you and @jsf9k want to overrule that.

I think anytime we make meaningful changes we are potentially breaking functionality (in this case, that of the script in `extras/`).  Certainly we'd want users to be able to say, "Hey, this script worked in 1.5.3 but not in 1.5.4" so I think a version bump makes sense here.

**Comment by mcdonnnj** [comment]
> > > @cisagov/team-ois I'm a bit hesitant about the version bump given that this PR does not change the Python package functionality at all. Thoughts?
> > 
> > 
> > I was on the fence about that also, but I told @bra1ncramp to do it. Since we are changing the way an included script works with this PR, I felt it was warranted, but honestly, I'm fine either way if you and @jsf9k want to overrule that.
> 
> I think anytime we make meaningful changes we are potentially breaking functionality (in this case, that of the script in `extras/`). Certainly we'd want users to be able to say, "Hey, this script worked in 1.5.3 but not in 1.5.4" so I think a version bump makes sense here.

I get that but my core rationale is that the file only exists if you clone the repository or download it manually. The `extras/` directory is not included in the wheel when you build the package for distribution which is, in my mind, what the version represents: the Python package.

**Comment by jsf9k** [comment]
> > > > @cisagov/team-ois I'm a bit hesitant about the version bump given that this PR does not change the Python package functionality at all. Thoughts?
> > > 
> > > 
> > > I was on the fence about that also, but I told @bra1ncramp to do it. Since we are changing the way an included script works with this PR, I felt it was warranted, but honestly, I'm fine either way if you and @jsf9k want to overrule that.
> > 
> > 
> > I think anytime we make meaningful changes we are potentially breaking functionality (in this case, that of the script in `extras/`). Certainly we'd want users to be able to say, "Hey, this script worked in 1.5.3 but not in 1.5.4" so I think a version bump makes sense here.
> 
> I get that but my core rationale is that the file only exists if you clone the repository or download it manually. The `extras/` directory is not included in the wheel when you build the package for distribution which is, in my mind, what the version represents: the Python package.

OK, that is a valid argument.  I'm fine either way, so I'm happy to defer to @mcdonnnj here.  @mcdonnnj - do you mind removing that version bump commit and force-pushing?

**Comment by dav3r** [comment]
Since the version bump was removed, I went ahead and deleted the pre-merge ("Finalize version") and post-merge ("Create a release") checklist items from the PR description now that they are no longer applicable.

**Comment by dav3r** [comment]
@mcdonnnj Just waiting on your final approval and then I will merge this.

**Review by jsf9k — APPROVED** [review]
_Approval intensifies!!_

**Comment by jsf9k** [comment]
> @mcdonnnj Just waiting on your final approval and then I will merge this.

Just turn on _auto-merge_!  It will merge once @mcdonnnj approves.

**Review by mcdonnnj — APPROVED** [review]
oops I knew I forgot to do something.

**Pull request merged**
